src/main/scala/com.snowplowanalytics.lrumap/CreateLruMap.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2012-2018 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.lrumap
+
+import cats.Id
+import cats.syntax.functor._
+import cats.effect.Sync
+
+/** `CreateLruMap` provides an ability to initialize the cache,
+  * which effect will `F`
+  *
+  * *WARNING*: Due to support of non-`Sync`/eager algebras,
+  * users should be super-careful about creating *values* of `F[_]`,
+  * as they won't have RT/lazy semantics and will be memoized
+  *
+  * This is NOT a type class, as it does not have the coherence
+  * requirement, but can be passed implicitly
+  */
+trait CreateLruMap[F[_], K, V] {
+  /** Create an LruMap within `F` effect
+    *
+    * @param size Max size before evicting
+    */
+  def create(size: Int): F[LruMap[F, K, V]]
+}
+
+object CreateLruMap {
```

Comment:
why not move everything to the LruMap companion object? so we can have `LruMap.create[F: Sync, K, V]` and `LruMap.unsafeCreate[Id, K, V]`?

Reply:
I'm easy about that. Just wanted to highlight that `CreateLruMap` and `LruMap` are quite independent entities. Former is a "capability" another is an interface.

Also `create` and `unsafeCreate` don't quite represent the nature of instances. Its not unsafe to create the map in `Id`, but it will be unsafe to use it along with functional combinators such as `traverse`

Reply:
for me it is, on the one hand you could create and get a `IO[LruMap[IO, K, V]]` and on the other hand `LruMap[Id, K, V]`

Reply:
Another problem is that I'd like to avoid such short methods for instances: https://nrinaudo.github.io/scala-best-practices/tricky_behaviours/implicit_shadowing.html

Reply:
with create/unsafeCreate on `LruMap` we wouldn't need implicits, would we?

Reply:
I see what you mean now. I'm trying to put this into current Iglu Client's API, where we need to decide which cache is going to be used

This is how it works right now:

```scala
/** Id and IO succeed, e.g. Option fails */
def parseConfig[F[_]: CreateLruMap](config: Json) = ???
```

I don't think it is possible to have similar API with non-implicit function, but happy for suggestions.

Reply:
ah ok, didn't really take into consideration how the api would be consumed, makes sense :+1: